### ISSUE TITLE
Zsh issue after the last update

### ISSUE THREAD

**Issue by rif** [ajeetdsouza/zoxide#94]
Without changing anything else after upgrading I get the folowing error:
```sh
❯ eval "$(zoxide init zsh)"
zsh: defining function based on alias `zi'
zsh: parse error near `()'
```

I am using zinit that has zi alias defined. (https://github.com/zdharma/zinit)

**Comment by ajeetdsouza** [comment]
The is happening because zoxide tries to declare a `zi` function for interactive queries, which is clashing with zinit's `zi` alias. How do you think this should be handled?

Temporarily, you can fix this by:
- Calling `unalias zi` before `zoxide init`. This will remove zinit's alias, and allow zoxide to initialize without any conflicts.
- Initializing zoxide before zinit. zinit will override zoxide's defined function, which means you won't be able to use the interactive `zi` command.
- If you'd like to use another prefix instead of `z`, calling `zoxide init --cmd j` which will define zoxide's functions as `j`, `ji`, etc.

**Comment by rif** [comment]
I chose unaliasing zi before zoxide init.

Thank you for your help and your great tool! :100:

**Comment by SanchithHegde** [comment]
I'd like to mention that the change which caused this (which I believe is [this commit](https://github.com/ajeetdsouza/zoxide/commit/d8a29f4367372859fe2a75c3b7cceab670f00ec8)), also broke the previous ability to switch to a directory interactively by using `z -i dir_name`, when the `--no-aliases` option is used. Can we please have that back?

If a PR has to be made, what are the necessary changes? Would moving existing `zi()` code to act on `z -i` and aliasing `zi` to `z -i` be a viable solution? And I'm assuming this has to be done for all the three shells (fish, posix and powershell) for keeping them similar?

**Comment by ajeetdsouza** [comment]
Hey @SanchithHegde,

The ability to use `z -i` was deliberately broken to provide certain guarantees about the result. However, your use case is valid, and I will expose an internal function that should allow you to create your own custom alias.

**Comment by SanchithHegde** [comment]
Great. Thank you!

**Comment by NightMachinary** [comment]
`zoxide init zsh --no-aliases` still creates `zi`. This is a bug, isn't it?

**Comment by ajeetdsouza** [comment]
@NightMachinary I'm currently reworking the `init` module entirely, which will fix your issue once released.

**Comment by ajeetdsouza** [comment]
@NightMachinary this is fixed in https://github.com/ajeetdsouza/zoxide/commit/f4525db02fde1efa77f56f9f50b27cc1830d4912.

`zoxide init --no-aliases` no longer defines `z` and `zi`. Instead, it relies on the user to create aliases as required:

```bash
alias z='__zoxide_z'
alias zi='__zoxide_zi'
# etc.
```

**Comment by ajeetdsouza** [comment]
Released in zoxide 0.5.0.
